### ISSUE TITLE
object-curly-spacing does not distinguish ES6 module from object declaration

### ISSUE THREAD

**Issue by gajus** [babel/babel-eslint#174]
`"babel-eslint": "^4.0.10"`

Rule:

```json
"object-curly-spacing": [
    2,
    "never"
],
```

Code:

```js
export * as ui from './ui';
export * as addressSearch from './addressSearch';
export * as userAddress from './userAddress';
```

Unexpected behaviour:

```
/foo/src/app/reducers/index.js
  1:7   error  There should be no space after 'export'  object-curly-spacing
  1:16  error  There should be no space before 'from'   object-curly-spacing
  2:7   error  There should be no space after 'export'  object-curly-spacing
  2:27  error  There should be no space before 'from'   object-curly-spacing
  3:7   error  There should be no space after 'export'  object-curly-spacing
  3:25  error  There should be no space before 'from'   object-curly-spacing
```

Expected behaviour:

Should not produce the above error messages.

**Comment by hzoo** [comment]
Thanks for reporting - Since this kind of export would be a [es7](https://github.com/leebyron/ecmascript-more-export-from) type thing (and eslint won't support any nonstandards except jsx and object rest/spread) this would need to be a rule rewrite in eslint-plugin-babel so make an issue there (or I can do it)
